### DISCUSSION TITLE
How To Read ChiaDB

### DISCUSSION THREAD

**Discussion by SametBasturkk** [Chia-Network/chia-blockchain#14233] [Support]
I want to get some information from chiadb full_blocks block.

How can I read blob or buffer type of value at the chiadb? 

Which path I should follow for convert to text or hex?

Byte32 > ? > ?

**Comment by kimsk** [comment]
You can check [block_store.py](https://github.com/Chia-Network/chia-blockchain/blob/main/chia/full_node/block_store.py#L333) for python code and SQL used for querying them. The code snippet pulled from my [notebook](https://github.com/kimsk/chia-concepts/blob/main/notebooks/misc/sqlite/full_blocks.ipynb) below should help too.

```python
import zstd

@with_db_connection
async def get_block_bytes_by_height_async(
    db, height: uint32
) -> Optional[str]:
    query = f"SELECT block FROM full_blocks WHERE height=?"
    async with db.execute(query, (height,)) as cursor:
        row = await cursor.fetchone()
        if row is not None:
            block_bytes = zstd.decompress(row[0])
            return block_bytes
        else:
            return None

async def get_block_by_height_async(height: uint32) -> Optional[FullBlock]:
    block_bytes = await get_block_bytes_by_height_async(height)
    if block_bytes is None:
        return None
    block = FullBlock.from_bytes(block_bytes)
    return block

```

**Reply by SametBasturkk** [reply]
Thanks I will take a look
